youtube_dl/compat.py

```diff
@@ -3200,6 +3200,18 @@ def compat_register_utf8():
     def compat_datetime_timedelta_total_seconds(td):
         return (td.microseconds + (td.seconds + td.days * 24 * 3600) * 10**6) / 10**6
 
+# optional decompression packages
+# PyPi brotli package implements 'br' Content-Encoding
+try:
+    import brotli as compat_brotli
+except ImportError:
+    compat_brotli = None
+# PyPi ncompress package implements 'compress' Content-Encoding
+try:
+    import ncompress as compat_ncompress
+except ImportError:
+    compat_ncompress = None
```

Comment:
Has this ever been observed in the wild? I'm against adding a whole new dependency for a use case that's never encountered.

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding indicates `compress` has been deprecated for decades

Reply:
It says:
> ...  this content-encoding is not used by many browsers today ...

Not by Mozilla, eg. One guy on the Web seemed to [want it](http://www.sqaforums.com/forums/archive/index.php/t-26299.html?X-ANT-WAF-Attempt=1), though it's hardly a yt-dl application and it was indeed more than two decades ago.

However as `compress` is in RFC 9110 and easy* to support, I included it.

It's a fairly trivial point since encoding is negotiated and `compress` is arguably a different case from `br` (Myspace vs Instagram). We would never have seen a bug report if some site offered `compress` but not `gzip` or `deflate`. You could surely be absolved if yt-dlp failed to reflect this feature, even given the slicker dependency system.

Separately, I wonder what sites do today if the client says `Accept-Encoding: identity` (so no encoding). Would that provoke/alleviate fingerprinting issues?

\* Considering negotiation, the PR doesn't **advertise** any optional encodings, though, since I forgot to include that change.